lando/k8s/README.md

```diff
@@ -0,0 +1,148 @@
+# Kubernetes Lando
+This module provides support for running Bespin jobs via a k8s cluster. This consists of two services. 
+- `k8s.lando` which listens for messages and runs k8s jobs while updating bespin-api
+- `k8s.watcher` which watches the k8s job listing and updates lando when jobs have completed, and logs failed jobs
+
+This requires bespin-api, postgres, and rabbitmq services to be running as outlined in [External services](#external-services) below.
+
+## Setup
+
+### k8s Cluster setup
+Connect to your k8s cluster.
+
+Create a project where the jobs will be run
+```
+oc new-project lando-job-runner
+```
+
+Create a service account that will be used by k8s.lando to create jobs.
+```
+oc create sa lando
+```
+
+Give this account admin priv
+```
+oc create rolebinding lando-binding --clusterrole=admin --serviceaccount=lando-job-runner:lando
+```
+
+Find the name of a token for this service account
+```
+oc describe sa lando
+```
+
+Determine the token value:
+```
+oc describe secret <tokename>
+```
+This value will need to be added to your k8s lando config file under `cluster_api_settings.token`.
+
+Create a DukeDS agent secret that will be used to stage data and save output.
+Create a file containing your ddsclient config named `ddsclient.conf`.
+Use this file to populate the DukeDS secret for your agent.
+```
+oc create secret generic ddsclient-agent --from-file=config=ddsclient.cred
+```
+
+Build the CWL workflow running image (calrissian)
+```
+oc create -f https://raw.githubusercontent.com/Duke-GCB/calrissian/master/openshift/BuildConfig.yaml
+oc create role pod-manager-role --verb=create,delete,list,watch --resource=pods
+oc create rolebinding pod-manager-default-binding --role=pod-manager-role --serviceaccount=lando-job-runner:default
+```
+
+Build the lando-util image that will be used for the stage data, organize output, and upload results jobs.
+```
+oc create -f https://raw.githubusercontent.com/Duke-GCB/lando-util/master/openshift/BuildConfig.yml
+```
+
+Create a persistent volume for holding system data matching the name in `run_workflow_settings.volume_claim_name` from the config file below.
+
+### Config file setup
+Create a config file named `k8s.config`.
+Example content:
+```
+work_queue:
+  host: TODO
+  username: TODO
+  password: TODO
+  worker_username: TODO
+  worker_password: TODO
+  listen_queue: TODO
+
+cluster_api_settings:
+  host: TODO
+  token: TODO
+  namespace: lando-job-runner
+  verify_ssl: False
+
+bespin_api:
+  token: TODO
+  url: TODO
+
+stage_data_settings:
```

Comment:
In https://github.com/Duke-GCB/bespin-api/pull/120, we moved details like this (which VM image to run and what resources to give it) into [VMSettings](https://github.com/Duke-GCB/bespin-api/blob/fde72af14007a279ec2cb438ba92b72a0b5cb920/data/models.py#L142-L160). Do you think it would work better to store these details in bespin-api instead?

Reply:
> Do you think it would work better to store these details in bespin-api instead?

Yes. The current values for requested cpu and memory are just random guesses and very likely to change. 

The images are currently built within the external cloud. Updating bespin-api will require a manual step to build the image with the appropriate tag on the external cloud.

The *Config objects in [jobmanager.py](https://github.com/Duke-GCB/lando/blob/master/lando/k8s/jobmanager.py) receive the job parameter so they can pull out the details from the vm settings once bespin-api supports them.

https://github.com/Duke-GCB/lando/blob/ee13d4a7a48fd7f237aae0f29d589aa2d46ad802/lando/k8s/jobmanager.py#L370-L383